code/game/objects/items/storage/boxes.dm

```diff
@@ -1357,3 +1357,141 @@
 	desc = "A bag containing fresh, dry coffee robusta beans. Ethically sourced and packaged by Waffle Corp."
 	beantype = /obj/item/reagent_containers/food/snacks/grown/coffee/robusta
 
+/obj/item/storage/box/ice_cream_carton
+	icon_state = "ice_cream"
+	icon = 'icons/obj/food/containers.dmi'
+	name = "Big Top plain ice cream carton"
+	desc = "A classic ice cream brand; this carton contains plain ice cream."
+
+	//What flavor will be inside the carton
+	var/ice_cream_flavor = /obj/item/reagent_containers/food/snacks/ice_cream_scoop
+
+/obj/item/storage/box/ice_cream_carton/Initialize(mapload)
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.max_items = 7
+	STR.set_holdable(list(/obj/item/reagent_containers/food/snacks/ice_cream_scoop))
+
+/obj/item/storage/box/ice_cream_carton/PopulateContents()
+	for(var/i in 1 to 7)
+		new ice_cream_flavor(src)
```

Comment:
It is crafted via activating cardboard which for some reason seems to bypass PopulateContents() because when I tested it, crafted ice cream cartons were empty.